src/main/webapp/app/tasks/local-task.service.spec.ts

```diff
@@ -52,4 +53,21 @@ describe('LocalTaskService', () => {
     expect(localStorage.getItem).toHaveBeenCalled();
     expect(localStorage.setItem).toHaveBeenCalled();
   });
+
+  it('should update the status of the task', () => {
+    const data = {id: 'db1c4afe-d358-4b14-acfd-0953e882a6b6', name: 'Test Task', isCompleted: false};
+    // when
+    taskService.update(id, data);
+
+    // then
+    expect(data).toEqual({id: 'db1c4afe-d358-4b14-acfd-0953e882a6b6', name: 'Test Task', isCompleted: true});
+  });
+
+  it('should update the local storage with pending tasks', () => {
+    // when
+    taskService.clearCompletedTasks(mockTask);
```

Comment:
this shouldn't compile because you need to use give it an array but it does because you did not give the parameter a type. Additionally it would be good if you give it at least one completed and one uncompleted tasks to test the functionality.

Reply:
Great. That's nice logic, Thanks. I really appreciate this.